pkg/storage/unified/sql/db/dbimpl/dbimpl.go

```diff
@@ -76,7 +76,10 @@ func newResourceDBProvider(grafanaDB infraDB.DB, cfg *setting.Cfg, tracer trace.
 	// as fallback, and as it uses a dedicated INI section, then keys are not
 	// prefixed with "db_"
 	getter := newConfGetter(cfg.SectionWithEnvOverrides("resource_api"), "db_")
-	fallbackGetter := newConfGetter(cfg.SectionWithEnvOverrides("database"), "")
+	fallbackConfig, err := sqlstore.NewDatabaseConfig(cfg, nil)
```

Comment:
This is the fix. Instead of parsing config again, we use `sqlstore.NewDatabaseConfig` which already has parsing code and understands "url" override too.